bitsv/network/services/bsvbookguarda.py

```diff
@@ -47,13 +42,20 @@ def get_address_info(cls, address):
     def get_balance(cls, address):
         r = requests.get(cls.MAIN_BALANCE_API.format(address), timeout=DEFAULT_TIMEOUT)
         r.raise_for_status()  # pragma: no cover
-        return r.json()
+        return int(r.json()['balance'])
 
     @classmethod
     def get_transactions(cls, address):
-        r = requests.get(cls.MAIN_ADDRESS_API.format(address), timeout=DEFAULT_TIMEOUT)
-        r.raise_for_status()  # pragma: no cover
-        return r.json()['transactions']
+        page = 1
+        while True:
+            r = requests.get(cls.MAIN_TX_PULL_API.format(address, page), timeout=DEFAULT_TIMEOUT)
+            r.raise_for_status()  # pragma: no cover
+            response = r.json(parse_float=Decimal)
+            for txid in response['txids']:
+                yield txid
+            if page == response['totalPages']:
+                break
+            page += 1
 
```

Comment:
I am worried that having this as a generator leads to an inconsistency in the return value when using the NetworkAPI (which requires all subordinate apis to have a consistent / normalized return value)

Reply:
I am guessing you want to use it as an iterator over a bitcom namespace or something... which is valid but the other apis don't give back paging information so maybe there needs to be two separate versions of this - 1 as a generator and the other for plugging into the standard normalized `get_transactions` call.

Open to suggestions but NetworkAPI needs to have a consistent return type no matter which underlying api it is using to get the job done.

Reply:
Or alternatively could look at `WhatsonchainNormalised` and do the same thing for `BSVBookGuardaAPI` to keep the generator but unroll it into a list in `BSVBookGuardaAPINormalised.get_transactions`

Reply:
I think `get_transactions()` should define the order of its results.  I suggest newest to oldest, and I hope service implementations that support paging would be intelligent enough to match this ordering for efficiency.

Then, add a keyword arg to control the maximum number of results:  `get_transactions(*, n=10)`.  The hope is that the default of 10 would be less that the paging size of any service implementation, and so avoids paging.

For now I'd punt on any paging API.  Reconsider when bitsv has a lot more users.

Reply:
I kind of agree... bitsv is not intended to be heavy-duty client software for pulling in thousands of txids (or indexing a bitcom namespace for example)... 

At least, I don't have the time to spend on bitsv right now to allow the scope to creep out...

I have pushed what I think is a reasonable compromise to the above: https://github.com/AustEcon/bitsv/commit/fca4767e69dda17dd1c3355f3dfb1432002e99d2

Basically it just always pages through all txids and returns a list of txids (which keeps the return type the same as all other APIs)...

This will obviously be very bad for thousands of txids but I don't think that usecase is appropriate anyway (at least I don't have the ability to support it right now and if I did, I wouldn't be using slow, clunky RESTAPIs for it anyway...)